### ISSUE TITLE
Ability to download the original video

### ISSUE THREAD

**Issue by fmnamado** [photoview/photoview#674]
**Describe the solution you'd like**

Hello,
I can download the web optimized video directly from Photoview.
Is it possible to add the possibility of download the original/high quality version also from Photoview?

**Comment by viktorstrate** [comment]
That is not possible, but it would be a nice addition.

**Comment by rdhir001** [comment]
Yes: it seems that small 4K-HDR size videos will play very well - I am having difficulty with large size files.

What do you mean by 'download'? - or do you mean viewing the file on your TV?

... In Partnership and Service, 

rdhir001

**Comment by fmnamado** [comment]
No, download the file.
When I am in a PC, in a browser, for photos I have the option of download the thumbnail, low rez, or ORIGINAL PHOTO.
In video I just have the option of downloading the "web optimized video" 

is is not very important, but if I have the option of downloading the original in photos, why not in videos also?

**Comment by rdhir001** [comment]
OK - then why not download original video from Google Photos where the original video resides?

Am I missing anything?

... In Partnership and Service,

rdhir001

**Comment by rdhir001** [comment]
Sorry - I thought you had PhotoView App for Smart TV:

***

https://www.photoviewapp.com

***

Regards.

... In Partnership and Service,

rdhir001

**Comment by fmnamado** [comment]
No problem but I would say you are on the wrong github?

This github is about photoview self hosted app as you can see here https://github.com/photoview and here https://photoview.github.io/

how you get up to this thread? :)

**Comment by rdhir001** [comment]
Being new to github, I first asked for help on the wrong thread.

Well, it was good to converse with you - you are thorough, and very responsive.

E . N . J . O . Y! a joyful life.

Regards.

~ rdhir001

**Comment by fmnamado** [comment]
oh OK no problem, I found out strange is this is a thread I began to ask for help, and you were asnwering or asking other thing.

No problem, hope you have also a good life :)

**Comment by DaCHack** [comment]
Funny thread here :laughing: 

I would like to +1 this feature request.
I have my files on a SMB-NAS that is mounted in the photoview container so I like to have only the converted files in usual access without the delay to my NAS, but if I select a single file or also a full album to be downloaded it would be good to have the option to go back to the originals and download them.

Same is true for videos that I want to show on a big screen: Once selected through the web interface I would love to play the original (being linked to and streamed from the NAS then would be fine).

**Comment by kkovaletp** [comment]
@fmnamado, @DaCHack, do you still have this issue when using the latest `master` version of Photoview?
I just tested this case with several source videos in different formats and resolutions, and for all of them I see the only "Original" and "Thumbnail" options to download on the sidebar.

If this is still the issue, please let me know what additional steps I should do or what exactly videos to have to reproduce the issue. If not, please also let me know, so I can close it as completed.

**Comment by viktorstrate** [comment]
@kkovaletp as I remember it "Original" in this case is actually a generated version to make sure it's web compatible.

I think it would be a nice addition to be able to download the original unmodified video files.

**Comment by viktorstrate** [comment]
Agree, I think it initially ended up like this because of the internal architecture of the code. This could certainly be improved.

**Comment by kkovaletp** [comment]
I've created the #967 to track this case. From my point of view, this is very important, as this is part of the core functionality of the product.

**Comment by kkovaletp** [comment]
Here is a detailed description of the issue:

**Describe the bug**
When you open a Sidebar with info about a media (picture or video), there is the Download section with the "Original" and "Thumbnail" options. I expected that the "Original" refers to the source media file, while actually, it refers to the converted media with the same resolution from the internal cache folder. This is not what "Original" means to me (and most of the people).

**Steps to reproduce the behavior:**

1. Open Sidebar for a media (image or video)
2. Check the "Download" section and find the "Original" row
3. Download the "original" media by clicking on the link and compare it with the source media file (by diff or any other tool, able to deal with binary files)
4. See the difference --> they are not the same file

**Expected behavior**
The "Original" should download the source media file.

**Ways to fix**
1st of all, we need to rename the "Original" to something better, like "High resolution", or something else (propose the wording). Then, we need to add the possibility of downloading the source media file from the User's media library.

**Comment by googollee** [comment]
I agree on a way to download the original media. More crazy idea, since we are working on the hardware acceleration, how about a way to convert videos on the fly?

**Comment by kkovaletp** [comment]
> how about a way to convert videos on the fly?

This would require a gaming videocard to be installed on the server for a home-use scenario (1-2 video streams at the same time) and a crypto-mining farm for up to 10 (or not a much bigger number) streams.

I'd say, this would be a good feature, but for a small group of our users. I expect that most of the users, who can enable HW acceleration, are running servers with embedded GPU or mid-level discrete GPU onboard. So, HW acceleration would be still beneficial for them to decrease scan job running time and power used, but not enough for real-time re-coding.

**Comment by googollee** [comment]
It's not such a high requirement for a home server. I convert and stream 4k h265 videos with jellyfin server on the fly, on a J4025 CPU. Most modern CPUs with hardware acceleration have the ability to convert 1 or 2 video streams on the fly.

I agree on that It's not a high-priority requirement for a photo gallery.
